core/charm/charmpath.go

```diff
@@ -51,7 +51,7 @@ func NewCharmAtPath(path string) (charm.Charm, *charm.URL, error) {
 }
 
 func isNotExistsError(err error) bool {
-	if os.IsNotExist(errors.Cause(err)) {
+	if os.IsNotExist(err) {
```

Comment:
We should, maybe not in this PR, move all `os.IsNotExist(x)` to `errors.Is(x, fs.ErrNotExist)`

`This function predates [errors.Is](https://pkg.go.dev/errors#Is). It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrNotExist).`

Reply:
I will write a patch for it as part of the scripts and re-run it over this PR. We might as well modernise at the same time. Good pickup.